### ISSUE TITLE
CA1862: Recommend case insensitive comparison overloads with patterns stored across statements

### ISSUE THREAD

**Issue by carlossanlop** [dotnet/roslyn-analyzers#6689]
Follow up for CA1862 after merging this PR: https://github.com/dotnet/roslyn-analyzers/pull/6662

This case was suggested in the following comment: https://github.com/dotnet/roslyn-analyzers/pull/6662#discussion_r1230457650

> Should we consider also detecting simple patterns across statements, such as below?
> 
> ```cs
> var loweredStr = str.ToLower();
> if (loweredStr.Contains(anotherStr))
> ```
> 
> We would need to check that we have exactly one reference to the intermediate temporary variable.
> 
> If yes, we already added another analyzer which implements a simple heuristic for detecting above patterns: https://github.com/dotnet/roslyn-analyzers/blob/main/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferStringContainsOverIndexOfAnalyzer.cs. If not for v1, but feel it may be a worthwhile enhancement.

**Comment by mavasani** [comment]
@carlossanlop Is this issue addressed with #6662?

**Comment by carlossanlop** [comment]
No. This was opened as a follow up from #6662 from feedback you gave me there.

**Comment by mavasani** [comment]
Oh sorry about that..
